lib/worker.js

```diff
@@ -86,13 +89,30 @@ async function getLocalLinter(basedir) {
         } else {
           linter = tslintDef;
         }
-        tslintCache.set(basedir, linter);
+        tslintCache.set(fileDir, linter);
         return resolve(linter);
       },
     );
   });
 }
 
+async function getNodePrefixPath() {
+  return new Promise((fulfill) => {
+    const npmCommand = process.platform === 'win32' ? 'npm.cmd' : 'npm';
+    ChildProcess.exec(
+      `${npmCommand} get prefix`,
+      { env: Object.assign(Object.assign({}, process.env), { PATH: getPath() }) },
+      (err, stdout, stderr) => {
+        if (err || stderr) {
+          throw (err || new Error(stderr));
```

Comment:
I haven't tried this before, does this bubble up correctly as if you used the reject of the promise?

Reply:
Yeah, I juggled with doing it this way vs. using `reject()` and opted for this mainly because the try/catch in `getLinter()` looked cleaner to me than the alternative "two-callbacks" method. I suppose it's really personal preference and I'll defer to you guys if you have a preference one way or the other. [It does work, though.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)

> If an error is thrown in the executor function, the promise is rejected. The return value of the executor is ignored.

Reply:
That's great! I totally agree it looks cleaner =]. I'll be adopting this style from now on I think.

Reply:
Wait, just looked at it a little closer, the `.exec()` callback isn't guaranteed to be in the executor function though right? It may have a different call stack.

Reply:
Y'know, you're right. It would *normally* work, but the fact that it's taking place in another process in this case means that needs to be a `Promise.catch()` rather than a regular `try/catch` block. I just tested it by changing `npm` to `nom` and with that change made, now it does catch the rejection properly. 👍 Will update in a second.